### PR TITLE
Fix document_lifecycle.md

### LINKED ISSUE #936: [BUG] Documentation error on document_lifecycle.md
### What is the bug?
The doc on [guides/document_lifecycle.md](https://github.com/opensearch-project/opensearch-js/blob/main/guides/document_lifecycle.md#setup) states that you can get the response of client.indices.exists as a boolean value but it's a http response, to get the boolean value you need to unpack it's "body" field.

### How can one reproduce the bug?
Execute the code on guides/document_lifecycle.md, it will never create an index as the truth value of a filled object is true so the creation will never happen.

### PR DESCRIPTION
**Pull request by eusousu**
unpack exists boolean value from response body

### Description

Fix a mistake on docs

### Issues Resolved

Fixes #936 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: guides/document_lifecycle.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Next, create an index named `movies` with the default settings:
 ```javascript 
 const index = 'movies';
 client.indices.exists({ index: index })
-  .then(exists => {
+  .then({ body: exists } => {
```
Comment (nhtruong) [inline guides/document_lifecycle.md:18]: This is not a proper syntax unfortunately.

File: guides/document_lifecycle.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Next, create an index named `movies` with the default settings:
...
     if (!exists) {
```
Comment (nhtruong) [inline guides/document_lifecycle.md:19]: I think we should simply access the body of the response (named exists) to get the boolean value:
`if(!exists.body) {`

Comment (nhtruong) [reply guides/document_lifecycle.md:19]: Then the overall fix would be:
```
client.indices.exists({ index: index })
  .then(exists => {
  if (!exists.body) {
    return client.indices.create({ index: index });
  }
}).then(response => console.log(`Index ${response.body.index} created`));
```

**Review by nhtruong — APPROVED** [review]

File: guides/document_lifecycle.md (hunk 2 above)
Comment (eusousu) [reply guides/document_lifecycle.md]: Okay! Made the change :)

**Review by nhtruong — APPROVED** [review]

**Pull request merged**